lib/siwapp/recurring_invoices.ex

```diff
@@ -57,4 +58,82 @@ defmodule Siwapp.RecurringInvoices do
   def change(%RecurringInvoice{} = recurring_invoice, attrs \\ %{}) do
     RecurringInvoice.changeset(recurring_invoice, attrs)
   end
+
+  @doc """
+  Given a recurring_invoice id, returns the amount of invoices that should
+  be generated
+  """
+  @spec invoices_to_generate(pos_integer()) :: integer
+  def invoices_to_generate(id) do
+    theoretical_number_of_inv_generated(id) - generated_invoices(id)
+  end
+
+  # Given a recurring_invoice id, returns the amount of invoices already generated related to that recurring_invoice
+  @spec generated_invoices(pos_integer()) :: integer
+  defp generated_invoices(id) do
+    Invoice
+    |> InvoiceQuery.number_of_invoices_associated_to_recurring_id(id)
+    |> Repo.one()
+  end
+
+  # Given a recurring_invoice id, returns the amount of invoices that
+  # should have been generated from starting_date until today, both included
+  @spec theoretical_number_of_inv_generated(pos_integer()) :: non_neg_integer()
+  defp theoretical_number_of_inv_generated(id) do
+    rec_inv = get!(id)
+    today = Date.utc_today()
+
+    max_date =
+      [today, rec_inv.finishing_date]
+      |> Enum.reject(&is_nil(&1))
+      |> Enum.sort(Date)
+      |> List.first()
+
+    number_using_dates =
+      number_of_invoices_in_between_dates(
+        rec_inv.starting_date,
+        rec_inv.period,
+        rec_inv.period_type,
+        max_date
+      )
+
+    max_ocurrences = rec_inv.max_ocurrences || number_using_dates
```

Comment:
Sugerencia sólo, lo que mejor veas, es que no soy muy fan del uso de `||`
```suggestion
    max_ocurrences = Map.get(rec_inv, :max_ocurrences)
    if max_ocurrences do
        min(number_using_dates, max_ocurrences)
    else
        number_using_dates
    end
```

Reply:
Como ya empieza a ser muy grande esta función, no se si la parte de obtener el number using dates lo metería en otra función